cmd/operator-sdk/new/cmd.go

```diff
@@ -436,3 +462,26 @@ func initGit() error {
 	log.Info("Run git init done")
 	return nil
 }
+
+func checkProject() error {
+	log.Info("Checking project")
+	switch projutil.DepManagerType(depManager) {
+	case projutil.DepManagerGoMod:
+		// Run "go build ./..." to make sure all packages can be built
+		// currectly. From "go help build":
+		//
+		//	When compiling multiple packages or a single non-main package,
+		//	build compiles the packages but discards the resulting object,
+		//	serving only as a check that the packages can be built.
+		opts := projutil.GoCmdOptions{
+			PackagePath: "./...",
+			Dir:         filepath.Join(projutil.MustGetwd(), projectName),
+		}
+		if err := projutil.GoBuild(opts); err != nil {
+			return err
+		}
+	}
+
+	log.Info("Check project successful.")
```

Comment:
In case of another dep manager we just succeed? :)

Reply:
Yes. We can add other validators in the future in general and for `dep`. If you think I should move the log statements so they do not execute unless modules are used, I can.